cmd/kubeops/internal/handler/handler.go

```diff
@@ -59,14 +62,27 @@ type Config struct {
 	ChartClient  chartUtils.Resolver
 }
 
-// NewInClusterConfig returns an internal cluster config replacing the token.
-func NewInClusterConfig(token string) (*rest.Config, error) {
-	config, err := rest.InClusterConfig()
-	if err != nil {
-		return nil, err
-	}
+// NewClusterConfig returns a copy of an in-cluster config with a custom token and/or custom cluster host
+func NewClusterConfig(inClusterConfig *rest.Config, token string, cluster string, additionalClusters map[string]AdditionalClusterConfig) (*rest.Config, error) {
+	config := rest.CopyConfig(inClusterConfig)
 	config.BearerToken = token
 	config.BearerTokenFile = ""
+
+	if cluster == defaultClusterName {
+		return config, nil
+	}
+
+	clusterConfig, ok := additionalClusters[cluster]
```

Comment:
I would rename `clusterConfig` (I thought it was a `rest.Config` but it seems to be short information we store). Maybe `clusterParams` or `clusterProperties`

Reply:
Renamed to `additionalCluster` given that it's coming from the `additionalClusters` map.